app/models/file_entry.ts

```diff
@@ -0,0 +1,57 @@
+import { DateTime } from "luxon";
+import { randomUUID } from "node:crypto";
+import type { UUID } from "node:crypto";
+
+import { BaseModel, column } from "@adonisjs/lucid/orm";
+
+export default class FileEntry extends BaseModel {
+  @column({ isPrimary: true })
+  declare id: UUID;
+
+  @column({ columnName: "file_extension" })
+  declare fileExtension: string;
+
+  @column.dateTime({ autoCreate: true })
+  declare createdAt: DateTime;
+
+  @column.dateTime({ autoCreate: true, autoUpdate: true })
+  declare updatedAt: DateTime;
+
+  get diskKey() {
+    return `${this.id}.${this.fileExtension}`;
+  }
+
+  public static createFileEntry(extname: string | undefined): FileEntry {
+    const fileEntry = new FileEntry();
+    fileEntry.id = randomUUID();
+    fileEntry.fileExtension =
+      extname !== undefined && extname.length > 0 ? extname : "bin";
+    return fileEntry;
+  }
+
+  public static async deleteByKeyAndReturnResult(
```

Comment:
since that other delete variant no longer exists (and very well, cause you should always be checking if the entry actually got deleted) we can give this function a saner name, such as `deleteByKey`